test/parallel/test-repl-empty.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const repl = require('repl');
+
+{
+  let evalCalledWithExpectedArgs = false;
+
+  const options = {
+    eval: common.mustCall((cmd, context) => {
+      // Assertions here will not cause the test to exit with an error code
+      // so set a boolean that is checked in process.on('exit',...) instead.
+      evalCalledWithExpectedArgs = (cmd === '\n');
+    })
+  };
+
+  const r = repl.start(options);
+
+  try {
+    // Empty strings should be sent to the repl's eval function
+    r.write('\n');
+  } finally {
+    r.write('.exit\n');
+  }
+
+  process.on('exit', () => {
+    assert(evalCalledWithExpectedArgs);
+  });
```

Comment:
I think you don’t need for the process to exit, the eval callback should be called synchronousl